hell/adventcalendar/2024/16/index.md

```diff
@@ -1,52 +1,74 @@
-author: "Maureen"
-author_bio: "Short intro about yourself"
+author: "Maureen Holland"
+author_bio: "Lead front-end web developer at silverorange.com. Passionate about accessibility, performance, and biking around Scotland."
 date: 2024-12-16
 author_links:
-  - label: "xy Blog"
-    url: "#"
-    link_label: "your-awesome-website.com/blog"
-  - label: "yx on ABC"
-    url: "https://abc-social-media-site"
-    link_label: "@xy"
+  - label: "Work"
+    url: "https://blog.silverorange.com/"
+    link_label: "blog.silverorange.com"
+  - label: "Personal"
+    url: "https://maureenholland.ca/magpie/"
+    link_label: "maureenholland.ca/magpie"
+  - label: "Mastodon"
+    url: "https://hachyderm.io/@maureenholland"
+    link_label: "@maureenholland"
 active: true
-intro: "<p>Short description of the post</p>"
+intro: "<p>How to get a quick accessibility win by repurposing the logic of applying a conditional class to provide meaningful semantics.</p>"
 image: "advent_16"
 ---
 
-Lorem ipsum dolor sit amet consectetur adipisicing elit. In aperiam alias laborum magnam accusamus accusantium vero. Exercitationem distinctio at quisquam quibusdam rem voluptatum illo, laboriosam natus aperiam sequi, voluptate inventore!
+Don’t get me wrong. You can keep it if you like it. But you don’t *need* it.
 
-## Lorem ipsum dolor sit amet.
+A class selector can allow us to visually show or hide content for disclosure widgets, like a custom select component or dropdown navigation menu. But a disclosure widget is made of two parts:
 
-Lorem ipsum dolor sit amet `consectetur` adipisicing elit. In aperiam alias laborum magnam accusamus accusantium vero. Exercitationem distinctio at quisquam quibusdam rem voluptatum illo, laboriosam natus aperiam sequi, voluptate inventore!
+1. A button that controls show/hide behaviour
+2. Related content that is shown or hidden *depending on the button state*
 
+Whether the `isOpen` class is applied or not, a screenreader would announce this button as **"Click me!, button"**.
 
 ```html
-<!DOCTYPE html>
-<html lang="en">
-  <head>
-	  <meta charset="utf-8" />
-	  <title>Webpage starter</title>
-  </head>
-  <body>
-    <h1>Hello World!</h1>
-  </body>
-</html>
+<button>Click me!</button>
+<p class="isOpen">Content that displays depending on a conditionally applied class.</p>
 ```
 
-## Lorem ipsum dolor sit amet.
+One of the principles of web accessibility is to be perceivable. That means not relying exclusively on one sense, like sight, to provide information. In the above case, if I can’t see the content being shown or hidden, how do I know my button click did anything?
 
-* Lorem, ipsum.
-* Lorem, ipsum.
-* Lorem, ipsum.
-* Lorem, ipsum.
+We *need* a corresponding semantic update to indicate what happened. This is what the `aria-expanded` attribute provides.
 
-![image](NSvIE.jpg)
+The following button would read: **"Click me!, expanded, button"**.
 
-or
+```html
+<button aria-expanded="true">Click me!</button>
```

Comment:
I went back and forth on whether to include `aria-controls` as well, but decided to focus on the attribute needed for styling

Reply:
`aria-controls` is pretty much useless and also not the topic of this post. So, I support your decision. :)

If you ask me, the button should be hidden if there's no JS. You could write about PE but I don't think it's necessary.

Reply:
Thanks for the feedback! It's amazing how much you can second-guess yourself even in a short article. Will leave this as-is.